v2/svid/jwtsvid/svid.go

```diff
@@ -56,6 +56,11 @@ func ParseAndValidate(token string, bundles jwtbundle.Source, audience []string)
 			return nil, wrapJwtsvidErr(errors.New("token header missing key id"))
 		}
 
+		// forbid tokens which have the `typ` header, which is not either "JOSE" or "JWT"
+		if typ, present := tok.Headers[0].ExtraHeaders[jose.HeaderType]; present && typ != "JOSE" && typ != "JWT" {
+			return nil, jwtsvidErr.New("token header type not equal to either JWT or JOSE")
```

Comment:
Will need to use the wrapJwtsvidErr function to wrap this. The zeebo error class have been removed since this PR was opened.